### ISSUE TITLE
crash while load_weights

### ISSUE THREAD

**Issue by dicky98** [walktree/libtorch-yolov3#21]
loading weight ...
terminate called after throwing an instance of 'c10::Error'
  what():  !is_variable() ASSERT FAILED at /pytorch/c10/core/TensorImpl.h:922, please report a bug to PyTorch. (shallow_copy_and_detach at /pytorch/c10/core/TensorImpl.h:922)
frame #0: std::function<std::string ()>::operator()() const + 0x11 (0x7efc2cd3f441 in /home/youyi/Documents/packages/libtorch/lib/libc10.so)
frame #1: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x2a (0x7efc2cd3ed7a in /home/youyi/Documents/packages/libtorch/lib/libc10.so)
frame #2: <unknown function> + 0x95063d (0x7efc2d8a863d in /home/youyi/Documents/packages/libtorch/lib/libtorch.so.1)
frame #3: torch::autograd::Variable::Impl::set_data(at::Tensor const&) + 0x268 (0x7efc2d8a4b58 in /home/youyi/Documents/packages/libtorch/lib/libtorch.so.1)
frame #4: at::Tensor::set_data(at::Tensor) + 0x61 (0x564f2799e459 in ./yolo-app)
frame #5: Darknet::load_weights(char const*) + 0x816 (0x564f279980c0 in ./yolo-app)
frame #6: main + 0x1a1 (0x564f279d670d in ./yolo-app)
frame #7: __libc_start_main + 0xe7 (0x7efc1de31b97 in /lib/x86_64-linux-gnu/libc.so.6)
frame #8: _start + 0x2a (0x564f2799262a in ./yolo-app)

based on pytorch 1.1.0

**Comment by liangzz1991** [comment]
when i used latest libtorch(1.1.0) encounter this problem you not only do:
Replace every "torch::TensorFromBlob"  to  "at::from_blob" as #12  ,
note that if you used GPU need repalce at::from_blob().to(torch::kCUDA)

also the problem of shallow_copy due to "set_data" function in Darknet.cpp when load weights,you can do :
//bn_imp->bias.set_data(bn_bias);
bn_imp->bias = bn_bias.detach();
There will be 6/7 totally.
Other minor problems due to the deprecated of function you can google it, i cannot remeber

**Comment by walktree** [comment]
#12  #4

**Comment by zhanghongruiupup** [comment]
> #12 #4

still error,although try this method. hope your reply
